upstairs/src/lib.rs

```diff
@@ -4401,14 +4396,62 @@ impl Upstairs {
         let next_id = downstairs.next_id();
         let mut cur_offset: usize = 0;
 
-        let mut dep = downstairs.active.keys().cloned().collect::<Vec<u64>>();
-        dep.sort_unstable();
+        /*
+         * Walk the downstairs work active list, and pull out all the active
+         * jobs (anything we have not submitted back to the guest). Construct
+         * a list of dependencies for this write based on the
+         * following rules:
+         *
+         * - ignore everything that happened before the last flush
+         * - writes have to depend on the last flush completing
+         * - any overlap of impacted blocks requires a dependency
+         *
+         * TODO: scanning backwards, any overlap of impacted blocks will
```

Comment:
Is this the TODO for improve dependencies in the future? 
Maybe start with "While scanning backwards, any ..."  I first read it as the TODO was `scanning backwards`.

Reply:
improved in 4691ec5